cmd/notation/list.go

```diff
@@ -55,9 +59,26 @@ func runList(command *cobra.Command, opts *listOpts) error {
 	}
 
 	// write out
-	for _, sigManifest := range sigManifests {
-		fmt.Println(sigManifest.Blob.Digest)
+	return output(manifestDesc.Digest, sigManifests, reference)
+}
+
+func output(digest digest.Digest, sigManifests []notationRegistry.SignatureManifest, reference string) error {
+	ref, err := registry.ParseReference(reference)
+	if err != nil {
+		return err
 	}
 
+	sigCount := len(sigManifests)
+	if sigCount > 0 {
+		// print title
+		fmt.Printf("%s/%s@%s\n", ref.Registry, ref.Repository, digest)
+		fmt.Printf("└── %s\n", notationRegistry.ArtifactTypeNotation)
+
+		for _, sigManifest := range sigManifests[:sigCount-1] {
+			// print each signature digest
+			fmt.Printf("    ├── %s\n", sigManifest.Blob.Digest)
+		}
+		fmt.Printf("    └── %s\n", sigManifests[sigCount-1].Blob.Digest)
```

Comment:
@yizha1 In the recent code refactoring according to the OCI Artifact Referrers API, we are no long be able to get the blob digest efficiently. Instead, should we just print out the digest of the signature manifest? /cc @FeynmanZhou

Reply:
> @yizha1 In the recent code refactoring according to the OCI Artifact Referrers API, we are no long be able to get the blob digest efficiently. Instead, should we just print out the digest of the signature manifest? /cc @FeynmanZhou

I am OK with it. This is more aligned with `oras discover` results. Then the list spec should be updated 
Existing
```shell
<registry>/<repository>@<digest>
└── application/vnd.cncf.notary.v2.signature
    ├──<digest_of_signature>
    └──<digest_of_signature>
```
New
```shell
<registry>/<repository>@<digest>
└── application/vnd.cncf.notary.v2.signature
    ├──<digest_of_signature_manifest>
    └──<digest_of_signature_manifest>
```

Reply:
Updated to digest_of_signature_manifest